### ISSUE TITLE
Weird split in JSX ternary

### ISSUE THREAD

**Issue by vjeux** [prettier/prettier#2807]
**Prettier Version:** master

**Code**

([playground](https://prettier.io/playground/#%7B%22content%22%3A%22if%20(1)%5Cnif%20(1)%5Cnif%20(1)%5Cnif%20(1)%5CnloadingOverlay%20%3D%20!this.state.coreData%5Cn%20%20.shouldShowLoadingOverlay%20%3F%20null%20%3A%20(%5Cn%20%20%3Cdiv%3E%5Cn%20%20%20%20%3CCenteredContainer%20fullHeight%3D%7Btrue%7D%20horizontal%3D%7Btrue%7D%20vertical%3D%7Btrue%7D%3E%5Cn%20%20%20%20%20%20%3CSpinner%20background%3D%5C%22light%5C%22%20size%3D%5C%22large%5C%22%20%2F%3E%5Cn%20%20%20%20%3C%2FCenteredContainer%3E%5Cn%20%20%3C%2Fdiv%3E%5Cn)%3B%5Cn%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D))

```js
if (1)
  if (1)
    if (1)
      if (1)
        loadingOverlay = !this.state.coreData
          .shouldShowLoadingOverlay ? null : (
          <div>
            <CenteredContainer
              fullHeight={true}
              horizontal={true}
              vertical={true}
            >
              <Spinner background="light" size="large" />
            </CenteredContainer>
          </div>
        );
```

The way it splits the condition is weird:

```js
        loadingOverlay = !this.state.coreData
          .shouldShowLoadingOverlay ? null : (
```

**Comment by suchipi** [comment]
Is the bug here the lack of indentation on the div, or the way it treats the null consequent?

**Comment by vjeux** [comment]
The fact that it breaks on the member expression instead of the ternary

**Comment by duailibe** [comment]
What is the expected output here?

```js
        loadingOverlay = !this.state.coreData.shouldShowLoadingOverlay ? null : (
```

This exceeds the printWidth

**Comment by vjeux** [comment]
I don't really know what the output should be, it just looks weird as it is right now.

**Comment by lots0logs** [comment]
@duailibe The expected output IMO is:
```
loadingOverlay = !this.state.coreData.shouldShowLoadingOverlay
  ? null
  : (
```
Thoughts?

**Comment by rattrayalex** [comment]
I agree with @lots0logs . IMO this indicates that our heuristics about when to enter "JSX Mode" for ternary printing is incorrect. 

Rather than deciding primarily based on whether a JSXElement is returned, it should be based on whether the ternary has an ancestor which is a JSXExpressionContainer (though things get a bit trickier inside props). 

I'm working on this as part of a larger change involving ternaries.
